tracer/src/Datadog.Trace/ContinuousProfiler/ProfilerStatus.cs

```diff
@@ -0,0 +1,73 @@
+// <copyright file="ProfilerStatus.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Runtime.InteropServices;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.ContinuousProfiler
+{
+    internal class ProfilerStatus
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ProfilerStatus));
+
+        private readonly bool _isProfilingEnabled;
+        private readonly object _lockObj;
+        private bool _isInitiliazed;
+        private IntPtr _engineStatusPtr;
+
+        public ProfilerStatus()
+        {
+            _isProfilingEnabled = EnvironmentHelpers.GetEnvironmentVariable(EnvironmentVariables.ProfilingEnabled)?.ToBoolean() ?? false;
+            Log.Information("Continuous Profiler is {IsEnabled}.", _isProfilingEnabled ? "enabled" : "disabled");
+            _lockObj = new();
+            _isInitiliazed = false;
+        }
+
+        public bool IsProfilerReady
+        {
+            get
+            {
+                if (!_isProfilingEnabled)
+                {
+                    return false;
+                }
+
+                EnsureNativeIsIntialized();
+                return _engineStatusPtr != IntPtr.Zero && Marshal.ReadByte(_engineStatusPtr) != 0;
+            }
+        }
+
+        private void EnsureNativeIsIntialized()
+        {
+            if (_isInitiliazed)
+            {
+                return;
+            }
+
+            lock (_lockObj)
+            {
+                if (_isInitiliazed)
+                {
+                    return;
+                }
+
+                _isInitiliazed = true;
+
+                try
+                {
+                    _engineStatusPtr = NativeInterop.GetProfilerStatusPointer();
+                }
+                catch (Exception e)
+                {
+                    Log.Warning("No profiler related feature(s) will be enabled. Failed to retrieve profiler status native pointer: {Reason}", e.Message);
```

Comment:
We should probably use the `Log.Warning` overload that takes the entire exception object.

Reply:
I did that at first, but the [system test](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=94167&view=logs&j=f9b5f215-18f7-5982-482b-42aaa98babdc&t=3f7875b6-f8c9-5740-4fc6-e671f428789e&l=139)  was failing because the profiler is [enabled](https://github.com/DataDog/system-tests/blob/main/utils/build/docker/set-system-tests-weblog-env.Dockerfile#L10) (which should not be the case for now).

For me this won't be the case anymore when we will have the tracer and profiler in the same `tar.gz` archive. But the `system tests` are failing for now and what would be the way to fix them.

Is there some exception I can add for my case ? or should set this flag to `false` ?

Reply:
An exception has been added for this case in the system tests (will be removed once we have the profiler with the tracer on linux)